rest-server/templates/src/main/scala/com/actionml/templates/navhinting/NavHintingDataset.scala

```diff
@@ -67,40 +65,37 @@ class NavHintingDataset(engineId: String) extends Dataset[NHEvent](engineId) wit
   }
 
 
-  def persist(event: NHEvent): Validated[ValidateError, NHEvent] = {
+  private def persist(event: NHEvent): Validated[ValidateError, NHEvent] = {
     try {
       event match {
         case event: NHNavEvent => // nav events enqued for each user until conversion
-
           val conversion = event.properties.conversion.getOrElse(false)
-          val unconvertedJourney = activeJourneysDAO.findOneById(event.entityId)
-          if(!conversion) { // store in the user journey queue
-            if (unconvertedJourney.nonEmpty) {
-              val updatedJourney = enqueueAndUpdate(event, unconvertedJourney)
-              if (updatedJourney.nonEmpty) { // existing Journey so updAte in place
-                activeJourneysDAO.save(updatedJourney.get)
-                Valid(true)
-              } // else the first event for the journey is a conversion so ignore
-            } else { // no persisted journey so create it
-              activeJourneysDAO.insert(
-                Journey(
-                  event.entityId,
-                  Seq(JourneyStep(event.targetEntityId, DateTime.parse(event.eventTime)))
+          Await.result(activeJourneysDAO.find("_id" -> event.entityId).flatMap { unconvertedJourney =>
```

Comment:
There is a logic error here. In the develop branch `findOnById` return `None` when the user-id, which is the key for this doc, does not exist. This means I will create a new record but the way you wrote it causes an exception to be thrown. Notice the `if (unconvertedJourney.nonEmpty)` logic above. 

We could add a method to the DAO or do this some other way. BTW this seems unsafe because the queue I create might be created by another thread even if using sync DB calls, right? There are other cases where I read the DB, do some complex computation to create a new doc then upsert it. These are all unsafe, right?

Also the `.save` is the same as `upsert` semantically, I believe. And it is more clear so maybe we should use that word for the function.